Docker.DotNet.BasicAuth/MaybeSecureString.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace Docker.DotNet.BasicAuth
+{
+    internal struct MaybeSecureString : IDisposable
+    {
+#if !netstandard
+        public SecureString Value;
+
+        public MaybeSecureString(string str)
+        {
+            var secureStr = new SecureString();
+
+            if (str.Length > 0)
+            {
+                foreach (char c in str)
+                {
+                    secureStr.AppendChar(c);
+                }
+            }
+
+            Value = secureStr;
+        }
+
+        public MaybeSecureString(SecureString value)
+        {
+            Value = value;
+        }
+
+        public void Dispose()
+        {
+            Value.Dispose();
```

Comment:
Whoops. I also think I need to make sure we copy the thing since this guy basically takes ownership.